climada/hazard/base.py

```diff
@@ -1887,21 +1871,11 @@ def from_hdf5(cls, file_name):
         hazard_kwargs = dict()
         with h5py.File(file_name, 'r') as hf_data:
             for (var_name, var_val) in haz.__dict__.items():
-                if var_name != 'tag' and var_name not in hf_data.keys():
+                if var_name not in hf_data.keys():
                     continue
                 if var_name == 'centroids':
                     hazard_kwargs["centroids"] = Centroids.from_hdf5(
                         hf_data.get(var_name))
-                elif var_name == 'tag':
-                    # legacy behavior: haz_type used to be part of the hazard tag
-                    if "haz_type" in hf_data.keys():
```

Comment:
Will this not make it impossible to read files created with previous versions of CLIMADA as the haz_type cannot be loaded?

Reply:
Good question! And if so it should show in the integration test, shouldn't it?

Reply:
If the integration tests would be complete yes. But they are not. Ideally, we should write one now. (actually a unit test).

Reply:
Actually it's not a problem: the hazard type is read below in lines
```python
    elif isinstance(var_val, np.ndarray) and var_val.ndim == 1:
        hazard_kwargs[var_name] = np.array(hf_data.get(var_name))
```

Reply:
The test for it is `climada.test.test_api_client.TestClient.test_get_hazard`.